microbial_thermodynamics/cell_growth_ODE.py

```diff
@@ -0,0 +1,143 @@
+"""Summary:
+This calculates rates of change of population growth, ribosome fraction and
+internal energy production.
+
+Desctiption:
+I am uncertain if the R in calculate_lam and the r in dr are the same thing (ribosome
+fraction). Should I find values for gam and R or write functions.
+"""  # noqa: D205
+
+from typing import Callable
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def calculate_r_star(a: float, om: float = 1e9, Q: float = 0.45) -> float:
+    """Calculates ideal ribosome fraction.
+
+    Args:
+       a: internal energy (ATP) concentration
+       om: Ribosome fraction half saturation constant
+       Q: housekeeping fraction
+    """
+    u = (a * (1 - Q)) / (a + om)
+    return u
+
+
+def calculate_lam(gam: float, R: float, m: float = 1e8, fb: float = 0.7) -> float:
+    """Calculates species growth.
+
+    Args:
+     gam: Effective translation elongation rate
+     R: Ribosome fraction
+     m: Cell mass
+     fb: Average fraction of ribosomes bound
+    """
+    u = (gam * fb * R) / m
+    return u
+
+
+def dN(tN: float, N: float, R: float, d: float = 0.1) -> float:
+    """Rate of change of population growth.
+
+    Args:
+     tN: time
+     lam: species’ growth rate
+     N: current population
+     d: rate of biomass loss.
+     R: ribosome fraction
+    """
+    lam = calculate_lam(1, R)  # 1 is just there untill i find a real value
+    udN = (lam - d) * N
+    return udN
+
+
+def dr(tr: float, t: float, a: float, r: float) -> float:
+    """Rate of change of ribosome fration.
+
+    Args:
+     tr: time
+     t: characteristic time scale for growth
```

Comment:
I would call this `growth_time_scale` or something similar. The equation for it is equation S12 in the SI of the original paper